tests/ctst/common/common.ts

```diff
@@ -85,3 +90,91 @@ Given('an object {string} that {string}', async function (this: Zenko, objectNam
         await putObject(this, objectName);
     }
 });
+
+When('the user tries to perform the current S3 action on the bucket {int} times with a {int} ms delay',
+    async function (this: Zenko, numberOfRuns: number, delay: number) {
+        this.setAuthMode('test_identity');
+        const action = {
+            ...this.getSaved<ActionPermissionsType>('currentAction'),
+        };
+        if (action.action.includes('Version') && !action.action.includes('Versioning')) {
+            action.action = action.action.replace('Version', '');
+            this.addToSaved('currentAction', action);
+        }
+        for (let i = 0; i < numberOfRuns; i++) {
+            // For repeated WRITE actions, we want to change the object name
+            if (action.action === 'PutObject') {
+                this.addToSaved('objectName', `objectrepeat-${Utils.randomString()}`);
+            } else if (action.action === 'CopyObject') {
+                this.addToSaved('copyObject', `objectrepeatcopy-${Utils.randomString()}`);
+            }
+            await runActionAgainstBucket(this, this.getSaved<ActionPermissionsType>('currentAction').action);
+            if (this.getResult().err) {
+                // stop at any error, the error will be evaluated in a separated step
+                return;
+            }
+            await Utils.sleep(delay);
+        }
+    });
+
+Then('the API should {string} with {string}', function (this: Zenko, result: string, expected: string) {
+    this.cleanupEntity();
+    const action = this.getSaved<ActionPermissionsType>('currentAction');
+    switch (result) {
+    case 'success':
+        if (action.expectedResultOnAllowTest) {
+            assert.strictEqual(
+                this.getResult().err?.includes(action.expectedResultOnAllowTest) ||
+                    this.getResult().stdout?.includes(action.expectedResultOnAllowTest) ||
+                    this.getResult().err === null, true);
+        } else {
+            assert.strictEqual(!!this.getResult().err, false);
+        }
+        break;
+    case 'fail':
+        assert.strictEqual(this.getResult().err?.includes(expected), true);
+        break;
+    }
+});
+
+Then('the operation finished without error', function (this: Zenko) {
+    this.cleanupEntity();
+    assert.strictEqual(!!this.getResult().err, false);
+});
+
+Given('an upload size of {int} B for the object {string}', async function (
+    this: Zenko,
+    size: number,
+    objectName: string
+) {
+    this.addToSaved('objectSize', size);
+    if (this.getSaved<boolean>('preExistingObject')) {
+        if (objectName) {
+            this.addToSaved('objectName', objectName);
+        } else {
+            this.addToSaved('objectName', `object-${Utils.randomString()}`);
+        }
+        await putObject(this, this.getSaved<string>('objectName'));
+    }
+});
+
+When('I PUT an object with size {int}', async function (this: Zenko, size: number) {
```

Comment:
```suggestion
When('i PUT an object with size {int}', async function (this: Zenko, size: number) {
```

Reply:
That said, we already have this function in 2.7: https://github.com/scality/Zenko/blob/310e350703b91e07a2bda01ce76aeb7c4269377a/tests/ctst/common/common.ts#L103
We might want to backport it to avoid duplication after waterfall ?

Reply:
Any reason to have a lowercase `i`?

Reply:
And for the other one, I will try, but the function is not standard with the internal putObject we have, that already implements file creation, and we'll need to add code related to flakiness

Reply:
I chose to keep the sentence, but call `addMultipleObjects` inside: the step must be a `When` as we retrieve the result, and expect things on it

Reply:
> Any reason to have a lowercase `i`?

Usually we have the first letter in lowercase because the sentence starts with When or And etc... but you are right, the I should be in uppercase mb...

Would you be able to modify the functions that starts with a lowercase one when integrating to other branches ? I see some of them like `When i restore..`, `When i run sorbetctl` but there might be some others

Reply:
I will try to do it yes

Reply:
(I won't guarantee the above sentence, to save time though :laughing: )